transaction.go

```diff
@@ -290,6 +291,91 @@ func (t *Transaction) SignalUpdatesApplied() error {
 	return nil
 }
 
+// append is a helper function to append updates to a transaction on which
+// SignalSetupComplete hasn't been called yet
+func (t *Transaction) append(updates []Update, done chan error) {
+	defer close(done)
+
+	// If there is nothing to append we are done
+	if len(updates) == 0 {
+		return
```

Comment:
Makes sense to return an error I think actually.

Reply:
Yeah sure. I just thought that appending nothing would be expected to be valid since golang's `append` also allows zero length slices to be used.

Reply:
hmm, I'm not sure I agree that it should return an error. It may seem weird to append 0 updates, but it can happen, just like how sometimes you have code that appends 0 elements. e.g.
```go
head, rest := updates[0], updates[1:]
t.Append(rest)
```
`rest` could be empty here. We don't want to force the caller to explicitly check for `len(rest) > 0` before calling `Append`.